project/graph_utils.py

```diff
@@ -0,0 +1,54 @@
+"""
+A set of methods for working with a graph.
+"""
+
+from typing import Tuple
+import cfpq_data
+import networkx as nx
+
+
+def get_graph_info(graph: nx.MultiDiGraph) -> Tuple[int, int, set]:
```

Comment:
В задании просили по имени графа получать информацию о нём. Поэтому принимать функция должна строку с именем графа из датасета.

Reply:
Можно оставить вспомогательные функци, но должна быть та, которая делает ровно то, что в задании написано.

Reply:
У меня приложение способно выводить информацию о графе по его имени. Для этого используется функция get_graph_info из модуля manager, которая в качестве одной из вспомогательных функций как раз использует приведённую Вами

В args хранится распаршенное имя графа из консольной строки
[```suggestion
def get_graph_info(args: Namespace) -> None:
```

```suggestion
info = graph_utils.get_graph_info(get_graph(args.graph_name))
```](https://github.com/Krekep/formal-lang-course/blob/b07d8f806320da4209cc36868e0a54d1db8634ef/project/manager.py#L21)

Или должна быть функция которая принимает именно строку?

Reply:
Да, я не увидел. Хорошо, давайте оставим так. Но тогда вопрос к функции get_graph
https://github.com/Krekep/formal-lang-course/blob/b07d8f806320da4209cc36868e0a54d1db8634ef/project/manager.py#L100
Тут можно просто вызвать функцию cfpq_data.download и потом cfpq_data.graph_from_csv без циклов. Что думаете?

Reply:
Исправил
https://github.com/Krekep/formal-lang-course/blob/5fc0de93a062c38d495d7280caf36f75112deaf8/project/manager.py#L112